lib/teleterm/clusters/cluster.go

```diff
@@ -50,13 +51,36 @@ type Cluster struct {
 	clusterClient *client.TeleportClient
 	// clock is a clock for time-related operations
 	clock clockwork.Clock
+	// Auth server features
+	Features *proto.Features
 }
 
 // Connected indicates if connection to the cluster can be established
 func (c *Cluster) Connected() bool {
 	return c.status.Name != "" && !c.status.IsExpired(c.clock)
 }
 
+// GetClusterFeatures returns a list of features enabled/disabled by the auth server
+func (c *Cluster) GetClusterFeatures(ctx context.Context) (*proto.Features, error) {
+	var authPingResponse proto.PingResponse
+
+	err := addMetadataToRetryableError(ctx, func() error {
+		proxyClient, err := c.clusterClient.ConnectToProxy(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer proxyClient.Close()
+
+		authPingResponse, err = proxyClient.CurrentCluster().Ping(ctx)
+		return trace.Wrap(err)
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return authPingResponse.ServerFeatures, nil
```

Comment:
Is it possible for the cluster to have access requests enabled but the user to not have access to them?

In the web app, I see that the features are scoped per user rather than per cluster which I believe is a good choice. Ultimately in our case we can use the same endpoint for both (it already returns info about currently logged in user), it's just a matter of how we approach that on the backend.

Reply:
I was working on LoggedInUser ACL related things but decided to separate it into it's own PR. Reading the comments here gave me a bit to think about so the state of that PR was pushed back a bit.

Reply:
For access requests specifically, on the webui all users can see the tab and the UI even if they have no access to Access Requests. Although I believe this is in more of a "TODO" state than a conscious design. 

The current change I have proposed in the frontend UI wouldn't be mutually exclusive to this idea. If the cluster doesn't have Access Requests enabled, then the UI shouldn't show the feature at all.

If we have more granular user ACL for different aspects of access requests, this would be tied into the smaller components of the UI. (such as disabling the `DELETE` button if they do not have access to DELETE)